components/scream/src/physics/shoc/shoc_compute_conv_time_shoc_length_impl.hpp

```diff
@@ -18,8 +18,7 @@ void Functions<S,D>::compute_conv_time_shoc_length(
   Scalar&       conv_vel,
   Scalar&       tscale)
 {
-  const Scalar tmp_val = (conv_vel > 0 ? conv_vel : 0);
-  conv_vel = std::pow(tmp_val, C::THIRD);
+  conv_vel = std::pow(ekat::impl::max(conv_vel, sp(0)), C::THIRD);
```

Comment:
Don't need `sp` here.

Reply:
When I have max(conv_vel, 0), I get a 
```
error: no matching function for call to ‘max(scream::shoc::Functions<float, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace> >::Scalar&, int)
```
I think that `std::max` doesn't allow for mixed types.

Reply:
Ah, my bad.